neutromeratio/plotting.py

```diff
@@ -0,0 +1,223 @@
+    nsamples - number of bootstrap samples to draw    
+    """
+
+
+    rmse_list = list()
+    rs_list = list()
+    for i in range(nsamples):
+        bootstrap_df = bootstrap_tautomer_exp_predict_results(df)
+        exp = bootstrap_df.experimental_values
+        pred = bootstrap_df.predicted_values
+        rmse_list.append(array_rmse(exp, pred))
+        rs_list.append(scs.pearsonr(exp, pred)[0])
+
+    rmse_array = np.asarray(rmse_list)
+    rs_array = np.asarray(rs_list)
+
+    rmse = array_rmse(df.experimental_values, df.predicted_values)
+    rs = scs.pearsonr(df.experimental_values, df.predicted_values)[0]
+
+    return (
+        BootstrapDistribution(rmse, rmse_array),
+        BootstrapDistribution(rs, rs_array),
+    )
+
+class BootstrapDistribution:
+    """Represents a bootstrap distribution, and allows calculation of useful statistics."""
+
+    def __init__(self, sample_estimate: float, bootstrap_estimates: np.array) -> None:
+        """
+        Parameters
+        ----------
+        sample_estimate - estimated value from original sample
+        bootstrap_estimates - estimated values from bootstrap samples
+        """
+        self._sample = sample_estimate
+        self._bootstrap = bootstrap_estimates
+        # approximation of δ = X - μ
+        # denoted as δ* = X* - X
+        self._delta = bootstrap_estimates - sample_estimate
+        self._significance: int = 16
+        return
+
+    def __float__(self):
+        return float(self._sample)
+
+    def empirical_bootstrap_confidence_intervals(self, percent: float):
+        """Return % confidence intervals.
+        
+        Uses the approximation that the variation around the center of the 
+        bootstrap distribution is the same as the variation around the 
+        center of the true distribution. E.g. not sensitive to a bootstrap
+        distribution is biased in the median.
+
+        Note
+        ----
+        Estimates are rounded to the nearest significant digit.
+        """
+        if not 0.0 < percent < 100.0:
+            raise ValueError("Percentage should be between 0.0 and 100.0")
+        elif 0.0 < percent < 1.0:
+            raise ValueError("Received a fraction but expected a percentile.")
+
+        a = (100.0 - percent) / 2
+        upper = np.percentile(self._delta, a)
+        lower = np.percentile(self._delta, 100 - a)
+        return self._sig_figures(self._sample, self._sample - lower, self._sample - upper)
+
+    def bootstrap_percentiles(self, percent: float):
+        """Return percentiles of the bootstrap distribution.
+        
+        This assumes that the bootstrap distribution is similar to the real 
+        distribution. This breaks down when the median of the bootstrap 
+        distribution is very different from the sample/true variable median.
+
+        Note
+        ----
+        Estimates are rounded to the nearest significant digit.
+        """
+        if not 0.0 < percent < 100.0:
```

Comment:
repeated from above -- may replace this with a `validate_percent` function or similar